compiler/rustc_abi/src/map.rs

```diff
@@ -0,0 +1,403 @@
+        let bool_abis = bool_abis.map(|result| result.unwrap().unwrap_or(false));
+        let [aapcs, cmse_nonsecure_entry] = bool_abis;
+        (
+            AbiMap {
+                c_proper,
+                system,
+                system_varargs,
+                rust_cold,
+
+                // arm...
+                aapcs,
+                cmse_nonsecure_entry,
+
+                // x86-ish
+                stdcall,
+                fastcall,
+                thiscall,
+                vectorcall,
+
+                ..arch_map
+            },
+            error_map,
+        )
+    }
+
+    // serialization
+    pub fn to_json_object(&self) -> JsonObject {
+        let mut json_obj = JsonObject::new();
+        json_obj.insert("system".to_owned(), self.system.as_str().to_owned().into());
+        json_obj
+            .insert("system-varargs".to_owned(), self.system_varargs.as_str().to_owned().into());
+        json_obj.insert("rust-cold".to_owned(), self.rust_cold.as_str().to_owned().into());
+        if let Some(abi) = self.efiapi {
+            json_obj.insert("efiapi".to_owned(), abi.as_str().to_owned().into());
+        }
+        if let Some(abi) = self.stdcall {
+            json_obj.insert("stdcall".to_owned(), abi.as_str().to_owned().into());
+        }
+        if let Some(abi) = self.fastcall {
+            json_obj.insert("fastcall".to_owned(), abi.as_str().to_owned().into());
+        }
+        if let Some(abi) = self.thiscall {
+            json_obj.insert("thiscall".to_owned(), abi.as_str().to_owned().into());
+        }
+        if let Some(abi) = self.vectorcall {
+            json_obj.insert("vectorcall".to_owned(), abi.as_str().to_owned().into());
+        }
+        if let Some(abi) = self.win64 {
+            json_obj.insert("win64".to_owned(), abi.as_str().to_owned().into());
+        }
+        if let Some(abi) = self.sysv64 {
+            json_obj.insert("sysv64".to_owned(), abi.as_str().to_owned().into());
+        }
+        if self.cmse_nonsecure_entry {
+            json_obj.insert("cmse-nonsecure-entry".to_owned(), JsonValue::Bool(true));
+        }
+        if self.aapcs {
+            json_obj.insert("aapcs".to_owned(), JsonValue::Bool(true));
+        }
+        if self.gpu_kernel {
+            json_obj.insert("gpu-kernel".to_owned(), JsonValue::Bool(true));
+        }
+        if self.ptx_kernel {
+            json_obj.insert("ptx-kernel".to_owned(), JsonValue::Bool(true));
+        }
+        if self.avr_interrupt {
+            json_obj.insert("avr-interrupt".to_owned(), JsonValue::Bool(true));
+        }
+        if self.msp430_interrupt {
+            json_obj.insert("msp430-interrupt".to_owned(), JsonValue::Bool(true));
+        }
+        if self.riscv_interrupt {
+            json_obj.insert("riscv-interrupt".to_owned(), JsonValue::Bool(true));
+        }
+        if self.x86_interrupt {
+            json_obj.insert("x86-interrupt".to_owned(), JsonValue::Bool(true));
+        }
```

Comment:
don't need to serialize the ones that are entirely arch-dependent... they probably don't need to be public, either?